src/libraries/Microsoft.PowerFx.Core/IR/CoercionMatrix.cs

```diff
@@ -351,5 +356,26 @@ private static CoercionKind GetToStringCoercion(DType fromType)
                 return CoercionKind.None; // Implicit coercion?
             }
         }
+
+        private static CoercionKind GetUntypedObjectCoercion(DType toType)
+        {
+            switch (toType.Kind)
+            {
+                case DKind.String:
+                    return CoercionKind.UntypedToText;
+                case DKind.Boolean:
+                    return CoercionKind.UntypedToBoolean;
+                case DKind.Number:
+                    return CoercionKind.UntypedToNumber;
+                case DKind.Date:
+                    return CoercionKind.UntypedToDate;
+                case DKind.Time:
+                    return CoercionKind.UntypedToTime;
+                case DKind.DateTime:
+                    return CoercionKind.UntypedToDateTime;
```

Comment:
I thought we were just implicitly doing conversion to String, Bool, Number, not also D/DT/T?

Reply:
We are also supporting Color/GUID as well. I have added support for those types.